alerts/apps_alerts.libsonnet

```
@@ -6,11 +6,35 @@
     prefixedNamespaceSelector: if self.namespaceSelector != null then self.namespaceSelector + ',' else '',
   },
 
+  // TODO move into libs
+  local wrap_rule_for_labels(rule) =
+    // Detect Kind of rule from name unless hidden `kind field is passed in the rule`
+    local kind =
+      if std.objectHas(rule, 'kind') then rule.kind
+      else if std.startsWith(rule.alert, 'KubePod') then 'pod'
+      else if std.startsWith(rule.alert, 'KubeStateful') then 'statefulset'
+      else if std.startsWith(rule.alert, 'KubeDeploy') then 'deployment'
+      else if std.startsWith(rule.alert, 'KubeDaemon') then 'daemonset'
+      else null;
+
+    local labels = { join_labels: $._config['%ss_join_labels' % std.toString(kind)], on_labels: [kind, 'namespace'], metric: 'kube_%s_labels' % std.toString(kind) };
+
+    // Failed to identify kind - return raw rule
+    if kind == null then rule
+    // No join labels passed in the config - return raw rule
+    else if std.length(labels.join_labels) == 0 then rule
+    // Wrap expr with join group left
+    else
+      rule {
+        expr: '(' + super.expr + ') * on (%s) group_left(%s) %s' % [std.join(',', labels.on_labels), std.join(',', labels.join_labels), labels.metric],
+      },
+
   prometheusAlerts+:: {
     groups+: [
       {
         name: 'kubernetes-apps',
-        rules: [
+        rules: [wrap_rule_for_labels(rule) for rule in self.rules_],
```

I noticed this file extensively uses `std.map`; perhaps it's not a bad idea to stay consistent with the rest of the file:

```suggestion
        rules: std.map(wrap_rule_for_labels, self.rules_),
```